entrypoint.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+from datetime import datetime
+import json
+import os
+import sys
+import syndicate
+from syndicate.utils import action_log, action_setoutput, job_getoutput, job_addoutput, get_posts, fronted, mark_syndicated_posts
+
+action_inputs = {
+    'silos': os.getenv('INPUT_SILOS').splitlines(),
+    'mark_as_syndicated': json.loads(os.getenv('INPUT_MARK_AS_SYNDICATED'))
+}
+
+posts = get_posts()
+if not posts:
+    action_log("No posts added or updated, nothing to do.")
+    action_setoutput("time", datetime.now())
+    sys.exit()
+
+# Do the thing.
+# Result set format:
+# {
+#     '<silo>': {
+#         'added': {
+#             'path/to/new_post': ( <silo post id>, <silo post URL> ),
+#             ...
+#         },
+#         'modified': {
+#             'path/to/updated_post': ( <silo post id>, <silo post URL> ),
+#             ...
+#         },
+#     },
+#     ...
+# }
+syndicated_posts = syndicate.elsewhere(posts, action_inputs['silos']) or {}
+action_setoutput('syndicated_posts', syndicated_posts)
+# Merge output with output of any previous runs
+job_addoutput(syndicated_posts)
+
+if action_inputs['mark_as_syndicated']:
+    action_log("Marking newly syndicated posts...")
+    ## NOTE
+    # If silos were provided, commit only the results of this step. In the case
+    # where no silos were provided, commit all job results so far.
+    #
+    # This allows us to bundle syndications into as few or many commits as we
+    # want in our workflows.
+    ##
+    if not action_inputs['silos']:
```

Comment:
will this key always exist? if not, you may want to use `action_inputs.get('silos')` which will return `None` when the key is not present, else you'll get a `KeyError`

Reply:
Good point. 👍 I'll confirm and update if necessary.

Reply:
The key will always exist. I'm setting it explicitly up here:

https://github.com/dabrady/syndicate/blob/edc91a3a485e81f789f183b7cee96f29a571c1fd/entrypoint.py#L9-L12